include/matx/operators/interp.h

```diff
@@ -0,0 +1,474 @@
+        value_type h_high = x_high - x_query;
+        
+        value_type t = h_low / h;
+        value_type s = h_high / h;
+
+        value_type v = s * v_low \
+          + t * v_high \
+          + (h  * (m_low * s - m_high * t) + v_diff * (t - s)) * t * s;
+
+        return v;
+      }
+
+
+    public:
+      __MATX_INLINE__ std::string str() const { return "interp1()"; }
+
+      __MATX_INLINE__ Interp1Op(const OpX &x, const OpV &v, const OpXQ &xq) : 
+        x_(x), 
+        v_(v),
+        xq_(xq)
+      {
+        MATX_ASSERT_STR(x_.Size(0) == v_.Size(0), matxInvalidSize, "interp: sample points and values must have the same size");
+      }
+      
+      static __MATX_INLINE__ constexpr __MATX_HOST__ __MATX_DEVICE__ int32_t Rank() 
+      { 
+        return OpXQ::Rank(); 
+      }
+
+      constexpr __MATX_INLINE__ __MATX_HOST__ __MATX_DEVICE__ index_t Size(int dim) const
+      {
+        return xq_.Size(dim);
+      }
+
+      template <typename ShapeType, typename Executor>
+      __MATX_INLINE__ void PreRun([[maybe_unused]] ShapeType &&shape, [[maybe_unused]] Executor &&ex) const {
+        
+        // Allocate temporary storage for spline coefficients
+        if constexpr (std::is_same_v<Method, InterpMethodSpline>) {
+          static_assert(is_cuda_executor_v<Executor>, "cubic spline interpolation only supports the CUDA executor currently"); 
+
+          int n = static_cast<int>(x_.Size(0)); // number of sample points
+
+          cuda::std::array<index_t, OpX::Rank()> m_shape{n};
+          detail::AllocateTempTensor(m_, std::forward<Executor>(ex), m_shape, &ptr_m_);
+
+          detail::tensor_impl_t<value_type, 1> d_tensor, dl_tensor, du_tensor; // Derivatives at sample points (spline only)
+          value_type *ptr_dl_ = nullptr;
+          value_type *ptr_d_ = nullptr;
+          value_type *ptr_du_ = nullptr;
+
+          detail::AllocateTempTensor(dl_tensor, std::forward<Executor>(ex), m_shape, &ptr_dl_);
+          detail::AllocateTempTensor(d_tensor, std::forward<Executor>(ex), m_shape, &ptr_d_);
+          detail::AllocateTempTensor(du_tensor, std::forward<Executor>(ex), m_shape, &ptr_du_);
+
+          // Fill tridiagonal system via custom operator
+          InterpSplineTridiagonalFillOp(dl_tensor,d_tensor, du_tensor, m_, x_, v_).run(std::forward<Executor>(ex));
+
+          // Solve tridiagonal system using cuSPARSE
+          cudaStream_t stream = ex.getStream();
+          cusparseHandle_t handle = nullptr;
+          cusparseStatus_t cusparse_status = cusparseCreate(&handle);
+          MATX_ASSERT(cusparse_status == CUSPARSE_STATUS_SUCCESS, matxCudaError);
+          cusparse_status = cusparseSetStream(handle, stream);
+          MATX_ASSERT(cusparse_status == CUSPARSE_STATUS_SUCCESS, matxCudaError);
+
+          size_t workspace_size = 0;
+          void* workspace = nullptr;
+          if constexpr (std::is_same_v<value_type, float>) {
+            cusparse_status = cusparseSgtsv2_bufferSizeExt(
```

Comment:
Does this need a check to make sure cusparse is installed? It's not a default currently.

Reply:
Are there checks elsewhere? I see
```
#include <cusparse.h>
```
in several other header files.

Reply:
I just sent the wrong place. Let me fix that comment...

Reply:
Ok ignore my comment. I forgot cusparse comes with CTK. I was thinking of cuDSS.